acados/ocp_nlp/ocp_nlp_cost_ls.c

```diff
@@ -793,11 +793,11 @@ void ocp_nlp_cost_ls_update_qp_matrices(void *config_, void *dims_,
 
         // tmp_ny = W * res
         blasfeo_dsymv_l(ny, 1.0, &model->W, 0, 0, &memory->res,
-                0, 0.0, &work->tmp_ny, 0, &work->tmp_ny, 0);
+                0, 0.0, &model->y_ref, 0, &work->tmp_ny, 0);
```

Comment:
I also came across the issue of having NaN propagate from the y vector into the z vector even if beta==0.
This has been fixed for dgemv and dsymv in a recent commit https://github.com/giaf/blasfeo/commit/0bdc0ceb0bc51ed78802620eabd7c805044ae07e
The single-precision has not been fixed yet, this is still WIP.

I would suggest to revert these changes in acados (in order to make the code less confusing) and update BLASFEO with the fix in the dgemv and dsymv routines.

Reply:
Is the issue appearing in other routines other than `dgemv` and `dsymv` ?

Reply:
I think there was also an issue related to a `dgemm` routine in GNSF.

I can revert those commits and update BLASFEO instead.

Just realized that the commit you pointed to is only on your `dev` branch.
Are you going to merge that soon?
I just tried updating BLASFEO to your `dev` branch, but I don't know if you have a compatible HPIPM version, or which one to use.

Reply:
I will merge dev into master in a matter of days at most, I'm almost done there.

If I remember correctly dgemm should not suffer from the same issue if there is NaN in the C matrix.
But if you say it does, I can double check.